### PR TITLE
adds doctorly head mirrors

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Adds head mirrors, those flip up mirrors doctors wear on their heads

Doctors can purchase them from their clothing vendor

image

Examine-more-ing people while wearing a head mirror lets you investigate their head organs, revealing tooth pills, tongues, eyes organs, and ear deafness. Similar to what the flashlight is capable of, but with bonus ear action.

## Why It's Good For The Game

An alternative headgear for doctors to wear, with some unique ghetto-treatment flavor (albeit already covered by flashlights, but that's authentic)

## Changelog

:cl: Melbert
add: Doctors can now get head mirrors from their clothes vendor, to complete the doctor outfit
/:cl:

### CODE REVIEW

**Comment by Ticklezcat** [comment]
10/10 not a beret.

**Review by Wallemations — APPROVED** [review]
Clearly hitting someone with a flash when they're wearing this should also flash the attacker :clueless:

File: code/modules/clothing/head/jobs.dm [hunk 1]
```diff
@@ -592,6 +592,83 @@
...
+	if(examiner.dir == examining.dir) // disallow examine from behind - every other dir is OK
+		examine_list += span_warning("You attempt to use your [name] to examine [examining]'s head better... but [examining.p_theyre()] facing the wrong way.")
+		return
+
+	var/obj/item/organ/internal/tongue/has_tongue = human_examined.get_organ_slot(ORGAN_SLOT_TONGUE)
+	var/obj/item/organ/internal/ears/has_ears = human_examined.get_organ_slot(ORGAN_SLOT_EARS)
+	var/obj/item/organ/internal/eyes/has_eyes = human_examined.get_organ_slot(ORGAN_SLOT_EYES)
+	var/pill_count = 0
+	for(var/datum/action/item_action/hands_free/activate_pill/pill in human_examined.actions)
+		pill_count++
```
Comment (Jacquerel) [inline code/modules/clothing/head/jobs.dm:636]: Really tiny change but we don't need this unless `!human_examined.is_mouth_covered()`

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -55,6 +55,13 @@
 
 	return null
 
+/mob/living/carbon/is_ears_covered()
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/carbon_defense.dm:58]: I don't like this proc being named as if it returns a boolean when it actually returns an `atom | null`, I know we set that precedent with the `is_antagonist` macros but it makes me sad.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/mob/living/carbon/carbon_defense.dm:58]: it's named after / follows the precedent of the other procs `is_eyes_covered` and `is_mouth_covered`

Comment (Jacquerel) [reply code/modules/mob/living/carbon/carbon_defense.dm:58]: sad

**Pull request merged**